src/converters/intrinsics.ts

```diff
@@ -251,6 +313,130 @@ export const fnEquals: Intrinsic = {
     },
 };
 
+/**
+ * Ref intrinsic resolves pseudo-parameters, parameter logical IDs or resource logical IDs to their values.
+ *
+ * If the argument to a Ref intrinsic is not a string literal, it may be another CF expression with intrinsic functions
+ * that needs to be evaluated first.
+ *
+ * See also:
+ *
+ * - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html
+ */
+export const ref: Intrinsic = {
+    name: 'Ref',
+    evaluate: (ctx: IntrinsicContext, params: Expression[]): Result<any> => {
+        if (params.length != 1) {
+            return ctx.fail(`Ref intrinsic expects exactly 1 param, got ${params.length}`);
+        }
+        const param = params[0];
+
+        // Although not part of the CF spec, Output values are passed through CDK tokens as Ref structures; therefore
+        // Pulumi Ref intrinsic receives them and has to handle them.
+        if (isOutputReprInstance(param)) {
+            return ctx.resolveOutput(<OutputRepr>param);
+        }
+
+        // Unless the parameter is a literal string, it may be another expression.
+        //
+        // CF docs: "When the AWS::LanguageExtensions transform is used, you can use intrinsic functions..".
+        if (typeof param !== 'string') {
+            const s = ctx.apply(ctx.evaluate(param), (p) => mustBeString(ctx, p));
+            return ctx.apply(s, (name) => evaluateRef(ctx, name));
+        }
+        return evaluateRef(ctx, param);
+    },
+};
+
+/**
+ * See `ref`.
+ */
+function evaluateRef(ctx: IntrinsicContext, param: string): Result<any> {
+    // Handle pseudo-parameters.
+    // See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html
+    switch (param) {
+        case 'AWS::AccountId':
+            return ctx.getAccountId();
+        case 'AWS::NoValue':
+            return ctx.succeed(undefined);
+        case 'AWS::Partition':
+            return ctx.getPartition();
+        case 'AWS::Region':
+            return ctx.getRegion();
+        case 'AWS::URLSuffix':
+            return ctx.getURLSuffix();
+        case 'AWS::NotificationARNs':
```

Comment:
I think this was actually a bug I introduced when I enabled use of StackId & StackName. `AWS::NotificationARNs` should still throw an error because it is not supported.

Reply:
I'll fix that!